### PR TITLE
Don't allow everything to be assignable to string within string mappings like Uppercase/Lowercase

### LINKED ISSUE #52731: `{} | unknown` is assignable to Uppercase<string> and Co.
# Bug Report

### 🔎 Search Terms

Intrinsic string manipulation types,Uppercase<string>, Lowercase<string>, Capitalize<string>, Uncapitalize<string>

### 🕗 Version & Regression Information

Since 4.8.4.
`Uppercase<string>` use to resolve to `string` between 4.1.5 and 4.7.4
The behaviour in 4.8.4 is not the same as in 4.9.5 / Nightly

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.9.5#code/C4TwDgpgBA8gRgKygXigbwL5QgD2BAOwBMBnKAVTEgCcBjAQxIgB4ThqBLAgcwD4oA-FHYBXaAC4oAM3oAbJgChQkCgQDWKKCPUEA9gHcC2PIVIUqEOoxZtOPfkNETpcpkA)

### 💻 Code

```ts
type Obj = {} extends Uppercase<string> ? true : false      // true
type Unk = unknown extends Uppercase<string> ? true : false // true
```

### 🙁 Actual behavior

The return value is `true`. It should not be, because intrinsic string manipulation types should resolve to subtypes of `string` (or `string` at the very least), and `{}` or `unknown` are not assignable to `string`.

### 🙂 Expected behavior

I expected the return type to be false.

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #52731

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot user test tsserver
@typescript-bot test tsserver top100
@typescript-bot run dt
@typescript-bot perf test this
@typescript-bot pack this

**Comment by jakebailey** [comment]
This PR isn't correct (note the test breaks), as it looks like some things only worked because of this return true.

But, given no real world code broke, I don't think I'm going to get any feedback for this besides unit testing.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -23872,10 +23872,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isMemberOfStringMapping(source: Type, target: Type): boolean {
-        if (target.flags & (TypeFlags.String | TypeFlags.Any)) {
+        if (target.flags & TypeFlags.Any) {
             return true;
         }
-        if (target.flags & TypeFlags.TemplateLiteral) {
+        if (target.flags & (TypeFlags.String & TypeFlags.TemplateLiteral)) {
```
Comment (ahejlsberg) [inline src/compiler/checker.ts:23878]: Surely you meant `TypeFlags.String | TypeFlags.TemplateLiteral`.

**Review by ahejlsberg — CHANGES_REQUESTED** [review]

File: src/compiler/checker.ts (hunk 1 above)
Comment (ahejlsberg) [reply src/compiler/checker.ts:23878]: Apparently we don't have a test for this case.

Comment (jakebailey) [reply src/compiler/checker.ts:23878]: Shoot, yes. Probably why the test fails.

Comment (jakebailey) [reply src/compiler/checker.ts:23878]: Bad copy paste, I had it the right way before!

**Review by ahejlsberg — APPROVED** [review]
Approved, provided the failing test runs are now clean.

**Comment by jakebailey** [comment]
@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot run dt
@typescript-bot perf test this

**Pull request merged**